core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1281,17 +1308,17 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
       // the `path exists` check for free
       if (zkClient.registerZNodeChangeHandlerAndCheckExistence(partitionReassignmentHandler)) {
         val partitionReassignment = zkClient.getPartitionReassignment
-        val partitionsToBeReassigned = partitionReassignment -- controllerContext.partitionsBeingReassigned.keys
-        partitionsToBeReassigned.foreach { case (tp, newReplicas) =>
-          if (topicDeletionManager.isTopicQueuedUpForDeletion(tp.topic)) {
-            error(s"Skipping reassignment of $tp since the topic is currently being deleted")
-            removePartitionFromReassignedPartitions(tp)
-          } else {
-            val reassignIsrChangeHandler = new PartitionReassignmentIsrChangeHandler(KafkaController.this, eventManager,
-              tp)
-            initiateReassignReplicasForTopicPartition(tp, ReassignedPartitionsContext(newReplicas, reassignIsrChangeHandler))
-          }
+        val newPartitionsToReassign = partitionReassignment -- controllerContext.partitionsBeingReassigned.keys
+
+        // Populate `partitionsBeingReassigned` with all partitions being reassigned before invoking
+        // `maybeTriggerPartitionReassignment` (see method documentation for the reason)
+        newPartitionsToReassign.foreach { case (tp, newReplicas) =>
+          val reassignIsrChangeHandler = new PartitionReassignmentIsrChangeHandler(KafkaController.this, eventManager,
+            tp)
+          controllerContext.partitionsBeingReassigned.put(tp, ReassignedPartitionsContext(newReplicas, reassignIsrChangeHandler))
```

Comment:
Again, this is not really introduced in this patch. However, since we only listen to the creation event of the partitionReassignment path, it seems that we can just initialize controllerContext.partitionsBeingReassigned with partitionReassignment?

Reply:
Yes. I've made the change.

One question is whether we should be defensive in the case that an operator deletes and recreates the `reassign_partitions` znode manually. There are probably multiple issues if that happens, but an obvious one is that we would end up with multiple registrations of ISR change handlers.